selfdrive/controls/controlsd.py

```diff
@@ -454,9 +454,13 @@ def state_transition(self, CS):
 
     # if stock cruise is completely disabled, then we can use our own set speed logic
     if not self.CP.pcmCruise:
-      self.v_cruise_kph = update_v_cruise(self.v_cruise_kph, CS.vEgo, CS.gasPressed, CS.buttonEvents,
-                                          self.button_timers, self.enabled, self.is_metric)
-      self.v_cruise_cluster_kph = self.v_cruise_kph
+      if CS.cruiseState.available:
```

Comment:
does this change cars where we don't need to set first, like Honda Nidec? I'd prefer to leave those as is

Reply:
We can try to have the car interfaces handle all the button logic and add a vCruise to CarState. For now added an exception until we have time for this clean up unless you want to do this now in another PR